### PR TITLE
HIVE-25795: [CVE-2021-44228] Update log4j2 version to 2.15.0

### PR DESCRIPTION
**Pull request by guptanikhil007**
### What changes were proposed in this pull request?

Update log4j2 version to 2.15.0

### Why are the changes needed?

To avoid Apache Log4j RCE Zero day exploit in all versions < 2.15.0

### CODE REVIEW

File: bin/hive-config.sh [hunk 1]
```diff
@@ -68,3 +68,7 @@ export HIVE_AUX_JARS_PATH=$HIVE_AUX_JARS_PATH
 
 # Default to use 256MB 
 export HADOOP_HEAPSIZE=${HADOOP_HEAPSIZE:-256}
+
+# Disable the JNDI. This feature has critical REC vulnerability.
+# when 2.x <= log4j.version <= 2.14.1
+export HADOOP_CLIENT_OPTS="$HADOOP_CLIENT_OPTS -Dlog4j2.formatMsgNoLookups=true"
```
Comment (dengzhhu653) [inline bin/hive-config.sh:74]: maybe we do not need this after updating?

**Comment by dengzhhu653** [comment]
++1 for this, maybe we should backport this for other hive branches.

File: bin/hive-config.sh (hunk 1 above)
Comment (guptanikhil007) [reply bin/hive-config.sh:74]: Similar HBase fix:
https://github.com/apache/hbase/pull/3933/files

The leader of knownsec 404 team (ZoomEye & SeeBug) 'Heige' have also recommend to set log4j2.formatMsgNoLookups to true
Ref: https://www.cyberkendra.com/2021/12/worst-log4j-rce-zeroday-dropped-on.html

Comment (dengzhhu653) [reply bin/hive-config.sh:74]: Got it, thank you!

**Pull request closed without merging**